core/services/vrf/v2/listener_v2_log_listener.go

```diff
@@ -0,0 +1,438 @@
+		minConfs = uint32(req.MinimumRequestConfirmations()) + uint32(lsn.job.VRFSpec.RequestedConfsDelay)
+	}
+	newConfs := uint64(minConfs) * (1 << lsn.respCount[req.RequestID().String()])
+	// We cap this at 200 because solidity only supports the most recent 256 blocks
+	// in the contract so if it was older than that, fulfillments would start failing
+	// without the blockhash store feeder. We use 200 to give the node plenty of time
+	// to fulfill even on fast chains.
+	if newConfs > 200 {
+		newConfs = 200
+	}
+	if lsn.respCount[req.RequestID().String()] > 0 {
+		lsn.l.Warnw("Duplicate request found after fulfillment, doubling incoming confirmations",
+			"txHash", req.Raw().TxHash,
+			"blockNumber", req.Raw().BlockNumber,
+			"blockHash", req.Raw().BlockHash,
+			"reqID", req.RequestID().String(),
+			"newConfs", newConfs)
+		vrfcommon.IncDupeReqs(lsn.job.Name.ValueOrZero(), lsn.job.ExternalJobID, lsn.coordinator.Version())
+	}
+	return req.Raw().BlockNumber + newConfs
+}
+
+func (lsn *listenerV2) handleFulfilled(fulfilled map[string]RandomWordsFulfilled) {
+	for _, v := range fulfilled {
+		// don't process same log over again
+		// log key includes block number and blockhash, so on re-orgs it would return true
+		// and we would re-process the re-orged request.
+		if !lsn.fulfillmentLogDeduper.ShouldDeliver(v.Raw()) {
+			continue
+		}
+		lsn.l.Debugw("Received fulfilled log", "reqID", v.RequestID(), "success", v.Success())
+		lsn.respCount[v.RequestID().String()]++
+		lsn.blockNumberToReqID.Insert(fulfilledReqV2{
+			blockNumber: v.Raw().BlockNumber,
+			reqID:       v.RequestID().String(),
+		})
+	}
+}
+
+func (lsn *listenerV2) handleRequested(requested []RandomWordsRequested, requestedLP []logpoller.Log, minConfs uint32) (pendingRequests []pendingRequest) {
+	for i, req := range requested {
+		// don't process same log over again
+		// log key includes block number and blockhash, so on re-orgs it would return true
+		// and we would re-process the re-orged request.
+		if lsn.inflightCache.Contains(req.Raw()) {
+			continue
+		}
+
+		confirmedAt := lsn.getConfirmedAt(req, minConfs)
+		lsn.l.Debugw("VRFListenerV2: Received log request",
+			"reqID", req.RequestID(),
+			"reqBlockNumber", req.Raw().BlockNumber,
+			"reqBlockHash", req.Raw().BlockHash,
+			"reqTxHash", req.Raw().TxHash,
+			"confirmedAt", confirmedAt,
+			"subID", req.SubID(),
+			"sender", req.Sender())
+		pendingRequests = append(pendingRequests, pendingRequest{
+			confirmedAtBlock: confirmedAt,
+			req:              req,
+			utcTimestamp:     requestedLP[i].CreatedAt.UTC(),
+		})
+		lsn.reqAdded()
+	}
+
+	return pendingRequests
+}
+
+// numReplayBlocks returns the number of blocks to replay on startup
+// given the request timeout and the chain ID.
+// if the chain ID is not recognized it assumes a block time of 1 second
+// and returns the number of blocks in a day.
+func numReplayBlocks(requestTimeout time.Duration, chainID *big.Int) int64 {
```

Comment:
I wonder if this belongs as a config to avoid requiring new releases. either TOML config (e.g. `EVM.Chain.AverageBlockTimeInSeconds`) or job-level config.

Reply:
I wanted to avoid adding job level configs because it'll make the migration harder, but maybe TOML config is best.

Reply:
Is it okay if we follow up with another PR for this config var?